### ISSUE TITLE
ioBroker not start by using latest installation command

### ISSUE THREAD

**Issue by SchumyHao** [ioBroker/ioBroker#113]
I want to install ioBroker on Macos.
I followed the step in this [link](https://github.com/ioBroker/ioBroker/wiki/Installation)

>node -v
>v8.12.0
>npm -v 
>6.5.0

After script finished, I can not open config UI.

The log shows

> 2019-01-25 18:45:12.529  - info: host.haojialeideMacBook-Pro.local iobroker.js-controller version 1.4.2 js-controller starting
> 2019-01-25 18:45:12.530  - info: host.haojialeideMacBook-Pro.local Copyright (c) 2014-2018 bluefox, 2014 hobbyquaker
> 2019-01-25 18:45:12.530  - info: host.haojialeideMacBook-Pro.local hostname: haojialeideMacBook-Pro.local, node: v8.12.0
> 2019-01-25 18:45:12.531  - info: host.haojialeideMacBook-Pro.local ip addresses: fe80::1490:7af:634:1e4 10.119.169.203 fe80::64c5:eff:fe60:dd0d fe80::92bd:2c2a:3efc:9aee fe80::847a:f8d1:84f4:dfbd
> 2019-01-25 18:45:12.559  - info: host.haojialeideMacBook-Pro.local  inMem-states listening on port 9000
> 2019-01-25 18:45:12.565  - info: host.haojialeideMacBook-Pro.local  inMem-objects listening on port 9001
> 2019-01-25 18:45:12.570  - info: host.haojialeideMacBook-Pro.local InMemoryDB connected
> 2019-01-25 18:45:12.573  - info: host.haojialeideMacBook-Pro.local no instances found

I successfully installed ioBroker before by using npm manually.

**Comment by AlCalzone** [comment]
Can you please post the contents of `INSTALLER_INFO.txt` in the iobroker directory (`/usr/local/iobroker`)?
CI tests seem to indicate that the installation should work on OSX, but I cannot test myself.

**Comment by AlCalzone** [comment]
Thanks. It seems the script fails to detect which init system is used. I'll have to investigate

**Comment by SchumyHao** [comment]
> Thanks. It seems the script fails to detect which init system is used. I'll have to investigate

If you need anything to debug on macos. I'm willing to help.

**Comment by SchumyHao** [comment]
After installation by [install.sh](https://raw.githubusercontent.com/ioBroker/ioBroker/stable-installer/installer.sh)

I have to run those two command and ioBroker can run finally

```
        iobroker del admin.0 && iobroker del discovery.0
        iobroker add admin && iobroker add discovery
```

**Comment by AlCalzone** [comment]
The docker issue is known and is related to the temporary hostnames. I have a PR up which should allow an easy fix.

Regarding OSX:  
If you can find out how to detect if `systemd` or `init.d` is used on OSX and how it is different from Linux, that would help a lot. Currently it is tested like this:
https://github.com/ioBroker/ioBroker/blob/master/installer.sh#L351

Unfortunately, I'm on holidays starting tomorrow, so I won't be able to do extended tests.

**Comment by AlCalzone** [comment]
> 
> 
> After installation by [install.sh](https://raw.githubusercontent.com/ioBroker/ioBroker/stable-installer/installer.sh)
> 
> I have to run those two command and ioBroker can run finally
> 
> ```
>         iobroker del admin.0 && iobroker del discovery.0
>         iobroker add admin && iobroker add discovery
> ```

That sounds like the docker issue. Once https://github.com/ioBroker/ioBroker.js-controller/pull/310 is merged, `iobroker host this` should fix it.
Can you please check which host the instances are assigned to directly after installation?
`iobroker object get system.adapter.admin.0 --pretty` => look for hostname

**Comment by AlCalzone** [comment]
It might be that this line is the problem:
https://github.com/ioBroker/ioBroker/blob/master/installer.sh#L352
Can you check if `/usr/local/etc/rc.d` exists on your system?

**Comment by Apollon77** [comment]
on my MacOS the path is not there

**Comment by Apollon77** [comment]
i think the old installer also was never capable of installing with autostart on MacOS because you need to do that completely different

**Comment by Apollon77** [comment]
I found e.g. this: https://medium.com/@fahimhossain_16989/adding-startup-scripts-to-launch-daemon-on-mac-os-x-sierra-10-12-6-7e0318c74de1

**Comment by AlCalzone** [comment]
I see. @SchumyHao would you be willing to prepare a PR to add autostart support to the installer script?

**Comment by SchumyHao** [comment]
> > After installation by [install.sh](https://raw.githubusercontent.com/ioBroker/ioBroker/stable-installer/installer.sh)
> > I have to run those two command and ioBroker can run finally
> > ```
> >         iobroker del admin.0 && iobroker del discovery.0
> >         iobroker add admin && iobroker add discovery
> > ```
> 
> That sounds like the docker issue. Once [ioBroker/ioBroker.js-controller#310](https://github.com/ioBroker/ioBroker.js-controller/pull/310) is merged, `iobroker host this` should fix it.
> Can you please check which host the instances are assigned to directly after installation?
> `iobroker object get system.adapter.admin.0 --pretty` => look for hostname


I want to debug this issue.
I removed /usr/local/iobroker and /usr/local/bin/iob*

Run install.sh again.
ioBroker dose not start automatically, so this issue can be reproduce. I think this is caused by macos's init system is different with linux.

But I do not need run
> > ```
> >         iobroker del admin.0 && iobroker del discovery.0
> >         iobroker add admin && iobroker add discovery
> > ```
I don't know why this issue can not be reproduce.

**Comment by Apollon77** [comment]
But you removed all iobroekr.files too or?

**Comment by SchumyHao** [comment]
> But you removed all iobroker-files too or?

As far as I know, files related with iobroker is located at /usr/local/iobroker and /usr/local/bin/iob* and I removed.

Can you show me where may contains other iobroker-files? Thank you @Apollon77

**Comment by SchumyHao** [comment]
About the autostart on OSX. I'm willing to help. I'll try to create PR when I complished.

**Comment by SchumyHao** [comment]
Hi @Apollon77 @AlCalzone 

I try to install this on Windows 10 buildin ubuntu enviroment.
I found ioBroker can not start even if I run `iobroker start`

The phenomenon is same as OSX.
No error log in log file.
And it recover after I run
```
iobroker del admin.0 && iobroker del discovery.0
iobroker add admin && iobroker add discovery
```

But I got more logs this time.
Before this issue recover, `iobroker host this` shows
```
Rename state "system.host.HAOJL2-PF182YX9.mem".
Rename host "system.host.HAOJL2-PF182YX9" to system.host.HAOJL2-PF182YX9
Rename state "system.host.HAOJL2-PF182YX9.memHeapUsed".
Rename state "system.host.HAOJL2-PF182YX9.memHeapTotal".
Rename state "system.host.HAOJL2-PF182YX9.memRss".
Rename state "system.host.HAOJL2-PF182YX9.uptime".
Rename state "system.host.HAOJL2-PF182YX9.load".
Rename state "system.host.HAOJL2-PF182YX9.alive".
Rename state "system.host.HAOJL2-PF182YX9.freemem".
Rename state "system.host.HAOJL2-PF182YX9.inputCount".
Rename state "system.host.HAOJL2-PF182YX9.outputCount".
```

When I run `iobroker del admin.0 && iobroker del discovery.0`
console shows
```
iobroker del admin.0 && iobroker del discovery.0

Delete adapter "admin.0"
host.HAOJL2-PF182YX9 no instances of adapter admin found
Delete adapter "discovery.0"
host.HAOJL2-PF182YX9 no instances of adapter discovery found
```
and `iobroker add admin && iobroker add discovery`
console shows
```
iobroker add admin && iobroker add discovery
host.HAOJL2-PF182YX9 install adapter admin
npm install --production (System call) in "/opt/iobroker/node_modules/iobroker.admin"
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm notice created a lockfile as package-lock.json. You should commit this file.

┌──────────────────────────────────────────────────────────┐
│                 npm update check failed                  │
│           Try running with sudo or get access            │
│           to the local update config store via           │
│ sudo chown -R $USER:$(id -gn $USER) /home/schumy/.config │
└──────────────────────────────────────────────────────────┘
got /opt/iobroker/node_modules/iobroker.admin/admin
upload [3] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/words.js words.js application/javascript
upload [2] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index_m.html index_m.html text/html
upload [1] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/index.html index.html text/html
upload [0] admin.admin /opt/iobroker/node_modules/iobroker.admin/admin/admin.png admin.png image/png
host.HAOJL2-PF182YX9 object system.adapter.admin created
host.HAOJL2-PF182YX9 create instance admin
host.HAOJL2-PF182YX9 object admin.0 created
host.HAOJL2-PF182YX9 object admin.0.info created
host.HAOJL2-PF182YX9 object system.adapter.admin.upload created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.outputCount created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.inputCount created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.uptime created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.memRss created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.memHeapTotal created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.memHeapUsed created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.connected created
host.HAOJL2-PF182YX9 object system.adapter.admin.0.alive created
host.HAOJL2-PF182YX9 object system.adapter.admin.0 created
host.HAOJL2-PF182YX9 install adapter discovery
npm install --production (System call) in "/opt/iobroker/node_modules/iobroker.discovery"
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated dgram@1.0.1: npm is holding this package for security reasons. As it's a core Node module, we will not transfer it over to other users. You may safely remove the package from your dependencies.
npm notice created a lockfile as package-lock.json. You should commit this file.

┌──────────────────────────────────────────────────────────┐
│                 npm update check failed                  │
│           Try running with sudo or get access            │
│           to the local update config store via           │
│ sudo chown -R $USER:$(id -gn $USER) /home/schumy/.config │
└──────────────────────────────────────────────────────────┘
got /opt/iobroker/node_modules/iobroker.discovery/admin
upload [5] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/words.js words.js application/javascript
upload [4] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/settings_m.html settings_m.html text/html
upload [3] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/settings.html settings.html text/html
upload [2] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/index_m.html index_m.html text/html
upload [1] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/index.html index.html text/html
upload [0] discovery.admin /opt/iobroker/node_modules/iobroker.discovery/admin/discovery.png discovery.png image/png
host.HAOJL2-PF182YX9 object system.adapter.discovery created
host.HAOJL2-PF182YX9 create instance discovery
host.HAOJL2-PF182YX9 object discovery.0.scanRunning created
host.HAOJL2-PF182YX9 object discovery.0.devicesFound created
host.HAOJL2-PF182YX9 object discovery.0.instancesFound created
host.HAOJL2-PF182YX9 object discovery.0.servicesProgress created
host.HAOJL2-PF182YX9 object discovery.0.devicesProgress created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.outputCount created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.inputCount created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.uptime created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.memRss created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.memHeapTotal created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.memHeapUsed created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.connected created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0.alive created
host.HAOJL2-PF182YX9 object system.adapter.discovery.0 created
```

I think it is **abnormal** that no instant found when I delete first time.

After this issue recover, `iobroker host this` shows
```
Rename state "system.host.HAOJL2-PF182YX9.mem".
Rename host "system.host.HAOJL2-PF182YX9" to system.host.HAOJL2-PF182YX9
Rename state "system.host.HAOJL2-PF182YX9.memHeapUsed".
Rename state "system.host.HAOJL2-PF182YX9.memHeapTotal".
Rename state "system.host.HAOJL2-PF182YX9.memRss".
Rename state "system.host.HAOJL2-PF182YX9.uptime".
Rename state "system.host.HAOJL2-PF182YX9.load".
Rename state "system.host.HAOJL2-PF182YX9.alive".
Rename state "system.host.HAOJL2-PF182YX9.freemem".
Rename state "system.host.HAOJL2-PF182YX9.inputCount".
Rename state "system.host.HAOJL2-PF182YX9.outputCount".
Instance host changed for "system.adapter.admin.0" from "HAOJL2-PF182YX9" to "HAOJL2-PF182YX9".
Instance host changed for "system.adapter.discovery.0" from "HAOJL2-PF182YX9" to "HAOJL2-PF182YX9".
```

**Comment by AlCalzone** [comment]
Can you please show me the instance objects right after installation as I asked you to?

**Comment by SchumyHao** [comment]
> Before this issue recover, `iobroker host this` shows
> 
> ```
> Rename state "system.host.HAOJL2-PF182YX9.mem".
> Rename host "system.host.HAOJL2-PF182YX9" to system.host.HAOJL2-PF182YX9
> Rename state "system.host.HAOJL2-PF182YX9.memHeapUsed".
> Rename state "system.host.HAOJL2-PF182YX9.memHeapTotal".
> Rename state "system.host.HAOJL2-PF182YX9.memRss".
> Rename state "system.host.HAOJL2-PF182YX9.uptime".
> Rename state "system.host.HAOJL2-PF182YX9.load".
> Rename state "system.host.HAOJL2-PF182YX9.alive".
> Rename state "system.host.HAOJL2-PF182YX9.freemem".
> Rename state "system.host.HAOJL2-PF182YX9.inputCount".
> Rename state "system.host.HAOJL2-PF182YX9.outputCount".
> ```

This is the instace objects right after installation is acomplished.

**Comment by AlCalzone** [comment]
No that is the host. I meant this
iobroker object get system.adapter.admin.0 --pretty
=> look for hostname

**Comment by Apollon77** [comment]
@SchumyHao We never tried installing in Linux subsystem on windows so far :-) So I also expect that autostart will not work there (I would be surprised else :-)
But very interesting exoeriment ... also because on Forum user tried it too these days :-)

For Windows we currently prepare a new native installer that installs everything

**Comment by SchumyHao** [comment]
> @SchumyHao We never tried installing in Linux subsystem on windows so far :-) So I also expect that autostart will not work there (I would be surprised else :-)
> But very interesting exoeriment ... also because on Forum user tried it too these days :-)
> 
> For Windows we currently prepare a new native installer that installs everything

I just preparing ioBroker doc in Chinese, so I'm trying all possible situations user may come across.

This issue is not so urgent. You can take your time to follow this issue. I'll update something if I find.

**Comment by frankjoke** [comment]
The old install did not work for me a year ago, but I apt-get updated my system and tried new installer now, it works with the exception of autostart. That's good for me because I do not want to start it automatically but only when I want to test something!

**Comment by AlCalzone** [comment]
@frankjoke which os are you using? And can you please post your INSTALLER_INFO.txt?
Although you don't want autostart, I want to find out why it doesn't work.

**Comment by frankjoke** [comment]
What I got on latest win10 1809 was:
`Installer version: 2019-01-25
Installation date 2019-01-28
Platform: linux
Installed as root
init system: unknown
Autostart: false`

**Comment by frankjoke** [comment]
[2019_01_28-15_32_58_backupiobroker.tar.gz](https://github.com/ioBroker/ioBroker/files/2803423/2019_01_28-15_32_58_backupiobroker.tar.gz)
I hope this backup gives you what you want!
Otherwise let me know how to get the data you need.

**Comment by frankjoke** [comment]
Al, installation is not really working aftzereards. The reason is that I cannot install ANY adapter! Get always something like below:
`$ ./iobroker add cloud  --host AZ-HP15
NPM version: 6.4.1
npm install iobroker.cloud --production --save --prefix "/opt/iobroker" (System call)
npm
 ERR! path /root/.npm/_logsnpm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall scandir

npm
 ERR! Error: EACCES: permission denied, scandir '/root/.npm/_logs'npm ERR!  { Error: EACCES: permission denied, scandir '/root/.npm/_logs'
npm ERR!   stack: 'Error: EACCES: permission denied, scandir \'/root/.npm/_logs\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm 
ERR!   syscall: 'scandir',npm ERR!   path: '/root/.npm/_logs' }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR!
 permissions of the file and its containing directories, or try runningnpm ERR! the command again as root/Administrator (though this is not recommended).

glob error { Error: EACCES: permission denied, scandir '/root/.npm/_logs'  stack: 'Error: EACCES: permission denied, scandir \'/root/.npm/_logs\'',
  errno: -13,
  code: 'EACCES',
  syscall: 'scandir',
  path: '/root/.npm/_logs' }

npm 
ERR! path /root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall open
npm ERR! Error: EACCES: permission denied, open '/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1'
npm ERR!  { Error: EACCES: permission denied, open '/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1'
npm ERR!
   cause: npm ERR!    { Error: EACCES: permission denied, open '/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1'
npm ERR!      errno: -13,
npm ERR!      code: 'EACCES',
npm ERR!      syscall: 'open',
npm ERR!      path: '/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1' },
npm ERR!   isOperational: true,
npm ERR!   stack: 'Error: EACCES: permission denied, open \'/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'open',
npm ERR!   path: '/root/.npm/_cacache/index-v5/2d/b8/230a91a37d7902484544fa924be439e7a1e6ddf8a230ba8c965ffd68e5e1' }

npm ERR! npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).

ERROR: host.AZ-HP15 Cannot install iobroker.cloud: 243
ERROR: process exited with code 25`

**Comment by AlCalzone** [comment]
It seems that the backup is using npm with sudo incorrectly. Or did you?
In any case, here's the solution
https://forum.iobroker.net/viewtopic.php?p=200327#p200327

Regarding your previous post, I  have no idea how to autostart something in the Linux subsystem in windows. So that's okay in my book.

**Comment by AlCalzone** [comment]
Also, can we please focus on the original issue? This was about the autostart on OSX and somehow we are now talking about npm issues in the Linux subsystem of Windows 10 🤔

**Comment by Apollon77** [comment]
Yes we should seperate noth topics. @SchumyHao / @frankjoke can one please open a new issue for the WIndows-stuff. Thank you!

**Comment by SchumyHao** [comment]
> No that is the host. I meant this
> iobroker object get system.adapter.admin.0 --pretty
> => look for hostname

Hi @AlCalzone 
I reproduced this issue.
`iobroker object get system.adapter.admin.0 --pretty` shows `not found`

**Comment by AlCalzone** [comment]
That is very strange. Can you post your objects.json instead or is it too large?

**Comment by Apollon77** [comment]
Worstcase as zip attachement

**Comment by SchumyHao** [comment]
[iobroker.tar.gz](https://github.com/ioBroker/ioBroker/files/2812425/iobroker.tar.gz)
I upload whole iobroker files that in /usr/local/iobroker

In case some files missed.

**Comment by AlCalzone** [comment]
Thank you, I will have a look at it tonight

**Comment by AlCalzone** [comment]
I just got around to looking at the files. There are indeed no instance objects. admin and discovery seem to have been installed though.

Is the issue still persisting after your autostart PR?

**Comment by SchumyHao** [comment]
This issue seems can be fixed with @ #118
